android/java/org/chromium/chrome/browser/app/BraveActivity.java

```diff
@@ -362,6 +363,15 @@ public void onResumeWithNative() {
             if (layout != null && layout.isWalletIconVisible()) {
                 updateWalletBadgeVisibility();
             }
+
+            // If a full screen custom tab was closed and bottom controls are enabled,
+            // show the bottom toolbar controls again
+            if (FullScreenCustomTabActivity.sIsFullScreenCustomTabActivityClosed
+                    && BottomToolbarConfiguration.isBraveBottomControlsEnabled()) {
+                layout.onBottomControlsVisibilityChanged(true);
+            }
+            // Reset the flag tracking whether a full screen custom tab was closed
+            FullScreenCustomTabActivity.sIsFullScreenCustomTabActivityClosed = false;
```

Comment:
I would set that flag to `false` in `onResume` inside `FullScreenCustomTabActivity` as well. There could be a case I suppose that it has been closed and re-opened without ever showing `BraveActivity`.